domain/application/state/storage.go

```diff
@@ -129,3 +484,525 @@ func (st *State) DetachStorage(ctx context.Context, storageUUID corestorage.UUID
 	//TODO implement me
 	return errors.New("not implemented")
 }
+
+func (st *State) getStorageDetails(ctx context.Context, tx *sqlair.TX, storageUUID corestorage.UUID) (storageInstance, error) {
+	inst := storageInstance{StorageUUID: storageUUID}
+	query := `
+SELECT &storageInstance.*
+FROM   storage_instance
+WHERE  uuid = $storageInstance.uuid
+`
+	queryStmt, err := st.Prepare(query, inst)
+	if err != nil {
+		return storageInstance{}, jujuerrors.Trace(err)
+	}
+
+	err = tx.Query(ctx, queryStmt, inst).Get(&inst)
+	if err != nil {
+		if !errors.Is(err, sqlair.ErrNoRows) {
+			return storageInstance{}, errors.Errorf("querying storage %q life: %w", storageUUID, err)
+		}
+		return storageInstance{}, errors.Errorf("%w: %s", storageerrors.StorageNotFound, storageUUID)
+	}
```

Comment:
Un-nest me, it's easier to read.

```suggestion
	if errors.Is(err, sqlair.ErrNoRows) {
		return storageInstance{}, errors.Errorf("querying storage %q life: %w", storageUUID, err)
	} else if err != nil {
		return storageInstance{}, errors.Errorf("%w: %s", storageerrors.StorageNotFound, storageUUID)
	}
```